lib/Target/Cpp/TranslateToCpp.cpp

```diff
@@ -375,9 +440,13 @@ static LogicalResult printModule(CppEmitter &emitter, ModuleOp moduleOp) {
 }
 
 static LogicalResult printFunction(CppEmitter &emitter, FuncOp functionOp) {
-  auto &blocks = functionOp.getBlocks();
-  if (blocks.size() != 1)
-    return functionOp.emitOpError() << "only single block functions supported";
+  // We need to forward-declare variables if the function has multiple blocks.
+  if (!emitter.forwardDeclaredVariables() &&
+      functionOp.getBlocks().size() > 1) {
+    return functionOp.emitOpError()
+           << "Need to enable variable forward declaration when emitting code "
+              "for function with multiple blocks";
```

Comment:
I think we could improve this error message, as it gets you

```
test/Target/control_flow.mlir:6:1: error: 'func' op Need to enable variable forward declaration when emitting code for function with multiple blocks
```

in the case you don't pass `-forward-declare-variables`.